beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -714,3 +714,61 @@ func (f *ForkChoice) ParentRoot(root [32]byte) ([32]byte, error) {
 	}
 	return n.parent.root, nil
 }
+
+// OnPayloadAttestationMessage processes a new aggregated payload attestation message.
+func (s *Store) OnPayloadAttestationMessage(
+	ctx context.Context,
+	payloadAttestation *ethpb.PayloadAttestation,
+) error {
+	data := payloadAttestation.Data
+	blockRoot := bytesutil.ToBytes32(data.BeaconBlockRoot)
+
+	// Check if the block exists in the store
+	node, ok := s.nodeByRoot[blockRoot]
+	if !ok {
+		return errors.New("beacon block root unknown")
+	}
+
```

Comment:
The spec contains more checks. These probably should be enforced in the caller function

```
    # Only update payload boosts with attestations from a block if the block is for the current slot and it's early
    if is_from_block and data.slot + 1 != get_current_slot(store):
        return
    time_into_slot = (store.time - store.genesis_time) % SECONDS_PER_SLOT
    if is_from_block and time_into_slot >= SECONDS_PER_SLOT // INTERVALS_PER_SLOT:
        return
```

Reply:
Hi @terencechain, thanks for reviewing, I still have to implement those I wrote this function just to get started

Reply:
added these checks